client/state/selectors/get-current-plan-purchase-id.js

```diff
@@ -0,0 +1,33 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { get, isNaN } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { getCurrentPlan } from 'state/sites/plans/selectors';
+
+/**
+ * Returns a site's current plan purchase ID or null if the site doesn't exist or the purchase ID
+ * is unknown.
+ *
+ * The ID returned by this selector is retrieved from the site plan. It is intended to provide a
+ * means of retrieving full purhcase information based on a site and it's plan information.
+ * Caution! It _does not_ retrieve the ID from a purchase.
```

Comment:
It's confusing to me too, that's why I added the comment 🙂 

This is sort of an implementation detail, but the thing is that we're finding an ID in `state.sites.plans[ siteId ]`, which appears to correspond to a purchase ID. We're not getting an ID from `state.purchases`.

I'm hoping to clarify this because devs are responsible for ensuring the correct data has been queried and is present, otherwise the selectors are worthless, and this siteplans/purchases has been confusing for me as I worked on this.

Does that help clarify? Do you think I can express that better in the comment?